### ISSUE TITLE
Region search: not returning SNPs

### ISSUE THREAD

**Issue by julie-sullivan** [intermine/intermine#1283]
From Deepak Unni:

I came across a bug with the Genomic region search tool where if I search for a set of coordinates, the tool will return all features that fall within that region with the exception of SNPs.

I looked into the log files for the SQL queries that are fired for the search and here is what I got:
```
SELECT DISTINCT a2_.id AS a1_, a2_.primaryIdentifier AS a6_, a2_.symbol AS a7_, a2_.class AS a8_, a3_.primaryIdentifier AS a9_, a5_.intermine_start AS a10_, a5_.intermine_end AS a11_ FROM SequenceFeature AS a2_, Chromosome AS a3_, Organism AS a4_, Location AS a5_ WHERE a2_.organismId = a4_.id AND a4_.shortName = 'B. taurus' AND a5_.featureId = a2_.id AND a5_.locatedOnId = a3_.id AND a3_.primaryIdentifier = 'GK000001.2' AND a2_.class IN ('org.intermine.model.bio.Exon', 'org.intermine.model.bio.Gene', 'org.intermine.model.bio.GeneFlankingRegion', 'org.intermine.model.bio.Indel', 'org.intermine.model.bio.MRNA', 'org.intermine.model.bio.SNP') AND a5_.locatedOnId = a5_.locatedOnId AND int4range(1, 100000) && int4range(a5_.intermine_start, a5_.intermine_end) ORDER BY a5_.intermine_start, a2_.id, a3_.primaryIdentifier, a5_.intermine_end LIMIT 1000
```
It seems that the range query  int4range(a5_.intermine_start, a5_.intermine_end) is causing this problem. 
If I were to replace it with int4range(a5_.intermine_start - 1, a5_.intermine_end) then the query returns all the SNPs as well.

I think this is similar to the issue #1108 (https://github.com/intermine/intermine/issues/1108) where int4range() doesn't work well if the start and end are of the same value (which is the case for SNPs).

I tested this with Intermine 1.6.5 on PostgreSQL 9.3.1.

**Comment by julie-sullivan** [comment]
rs35149808 on humanmine is not returned by the region search correctly.     3:149187990-149187990

**Comment by deepakunni3** [comment]
I wasn't sure if this bug has been fixed and is being prepared for InterMine 1.6.6 but I thought I would chime in with a few things that I found.

`int4range()` has a constructor function, in PostgreSQL, that takes 3 arguments: upper-bound, lower-bound and text argument indicating inclusivity/exclusivity of bounds.
i.e. instead of `int4range(149187990, 149187990)`, it should be `int4range(149187990, 149187990, '[]')` to return a SNP that has position 149187990 on a chromosome 3 (using the example commented previously).

This inclusivity of bounds is critical for Genomic Region Search tool to find any feature, overlapping a given range, that has the same start and end (most commonly SNPs).

The easiest way to introduce this change was in `overlapConstraintToString()` function in `SqlGenerator.java`. Specifically, at Line [2070](https://github.com/intermine/intermine/blob/8f276ec401e6db779a354dd3c4896671b1082040/intermine/objectstore/main/src/org/intermine/objectstore/intermine/SqlGenerator.java#L2070) by adding `buffer.append(", '[]'")` right before closing the `int4range` statement.

I have had success in getting the Genomic Region Search tool to work as expected but I wasn't sure if this would have any downstream consequences. Would like to hear from the developers if there is a better way to address this or if it has already been addressed and the code is on a branch.
